src/rust/engine/src/nodes.rs

```diff
@@ -864,10 +864,21 @@ impl Task {
           .ok_or_else(|| throw(&format!("no edges for task {:?} exist!", entry)))
           .and_then(|edges| {
             edges.entry_for(&dependency_key).cloned().ok_or_else(|| {
-              throw(&format!(
-                "{:?} did not declare a dependency on {:?}",
-                entry, dependency_key
-              ))
+              let is_union = get.declared_subject.map(externs::is_union).unwrap_or(false);
+              if is_union {
+                let union_ty = get.declared_subject.unwrap();
```

Comment:
This is logically safe, but reading an `unwrap()` here is jarring requiring a secind look to make sure this can never error. What do you think about using an early return instead?:
```diff
diff --git a/src/rust/engine/src/nodes.rs b/src/rust/engine/src/nodes.rs
index 145ed93af..424b23b11 100644
--- a/src/rust/engine/src/nodes.rs
+++ b/src/rust/engine/src/nodes.rs
@@ -864,21 +864,20 @@ impl Task {
           .ok_or_else(|| throw(&format!("no edges for task {:?} exist!", entry)))
           .and_then(|edges| {
             edges.entry_for(&dependency_key).cloned().ok_or_else(|| {
-              let is_union = get.declared_subject.map(externs::is_union).unwrap_or(false);
-              if is_union {
-                let union_ty = get.declared_subject.unwrap();
-                let value = externs::get_value_from_type_id(union_ty);
-                let description = externs::project_str(&value, "union_description");
-                throw(&format!(
-                  "Type {} is not a member of the {} @union (\"{}\")",
-                  get.subject, union_ty, description
-                ))
-              } else {
-                throw(&format!(
-                  "{:?} did not declare a dependency on {:?}",
-                  entry, dependency_key
-                ))
+              if let Some(subject_type) = get.declared_subject {
+                if externs::is_union(subject_type) {
+                  let value = externs::get_value_from_type_id(subject_type);
+                  let description = externs::project_str(&value, "union_description");
+                  return throw(&format!(
+                    "Type {} is not a member of the {} @union (\"{}\")",
+                    get.subject, subject_type, description
+                  ))
+                }
               }
+              throw(&format!(
+                "{:?} did not declare a dependency on {:?}",
+                entry, dependency_key
+              ))
             })
           });
         // The subject of the get is a new parameter that replaces an existing param of the same
```
This would be really clean with the `let_chains` feature and a clear win with no need to nest ifs and early return, but maybe preferrable anyhow.

Reply:
I ended up refactoring this into a guarded match expression, which avoids both the `unwrap()` call and the early return